### PR TITLE
feat(cli): add `intentUrl` to `json` and `ndjson` reporters

### PR DESCRIPTION
**Pull request by ricokahler**
### Description

- Adds `intentUrl` to the results of `sanity documents validate --format json` and `--format ndjson`
- Fixes an edge case where there is an unnecessary leading slash in the intentUrl
- Fixes the logic in `validateDocument.test.ts` that failed to catch the documents with `system.` types to show up bug.
- Fixes a bug where valid documents would be reported in `sanity documents validate --format json`

### What to review

Take a look at the logic and ensure nothing else changes.

### Testing

- I ran the build and tried out the command for myself.

### Notes for release

- Includes `intentUrl` in `json` and `ndjson` formats for `sanity documents validate`
- Fixes a bug where valid documents would be reported in `sanity documents validate --format json`

### CODE REVIEW

File: packages/sanity/src/_internal/cli/threads/__tests__/validateDocuments.test.ts [hunk 1]
```diff
@@ -59,11 +59,11 @@ const documents: SanityDocument[] = [
     },
   },
   {
-    _id: 'system.some-system-document.foo',
-    _rev: 'rev6',
+    _id: 'some-system-document.foo',
```
Comment (ricokahler) [inline packages/sanity/src/_internal/cli/threads/__tests__/validateDocuments.test.ts:62]: this `_id` is why the test still passed lol

**Review by binoy14 — APPROVED** [review]
